src/components/textarea/textarea.test.jsx

```diff
@@ -0,0 +1,227 @@
+    });
+  });
+
+  describe('onBlur API', () => {
+    it('sets the onblur handler', () => {
+      const onBlurSpy = jest.fn();
+      render(<Textarea {...requiredProps} onBlur={onBlurSpy} />);
+      fireEvent.blur(screen.getByLabelText('the label'));
+      expect(onBlurSpy.mock.calls.length).toEqual(1);
+    });
+  });
+
+  describe('onChange API', () => {
+    it('sets the onchange handler', () => {
+      const onChangeSpy = jest.fn();
+      render(<Textarea {...requiredProps} onChange={onChangeSpy} />);
+      fireEvent.change(screen.getByLabelText('the label'), { target: { value: 'new value' } });
+      expect(onChangeSpy.mock.calls.length).toEqual(1);
+    });
+  });
+
+  describe('placeholder API', () => {
+    it('sets the placeholder attribute', () => {
+      render(<Textarea {...requiredProps} placeholder="test-placeholder" />);
+      expect(screen.getByLabelText('the label')).toHaveAttribute('placeholder', 'test-placeholder');
+    });
+  });
+
+  describe('readOnly API', () => {
+    it('sets the readOnly attribute', () => {
+      render(<Textarea {...requiredProps} readOnly />);
+      expect(screen.getByLabelText('the label')).toHaveAttribute('readonly');
+    });
+  });
+
+  describe('ref API', () => {
+    it('can be set', () => {
+      const ref = React.createRef();
+      render(<Textarea {...requiredProps} ref={ref} value="test value" />);
+      expect(ref.current.value).toEqual('test value');
+    });
+  });
+
+  describe('required API', () => {
+    it('can be set', () => {
+      render(<Textarea {...requiredProps} required={true} />);
+      expect(screen.getByLabelText('the label')).toBeRequired();
+      expect(screen.getByLabelText('the label')).toBeInvalid();
+    });
+
+    it('can be unset', () => {
+      render(<Textarea {...requiredProps} required={false} />);
+      expect(screen.getByLabelText('the label')).not.toBeRequired();
+      expect(screen.getByLabelText('the label')).toBeValid();
+    });
+
+    it('is invalid on mount but does not have an error message', () => {
+      render(<Textarea {...requiredProps} required={true} />);
+      expect(screen.getByLabelText('the label')).toBeInvalid();
+      expect(screen.getByLabelText('the label')).toHaveDescription('');
+    });
+
+    it('is invalid after typing', async () => {
+      render(<Textarea {...requiredProps} required={true} />);
+      userEvent.type(screen.getByLabelText('the label'), 'a');
+      expect(screen.getByLabelText('the label')).toHaveValue('a');
+      expect(screen.getByLabelText('the label')).toBeValid();
+      expect(screen.getByLabelText('the label')).toHaveDescription('');
+      userEvent.type(screen.getByLabelText('the label'), '{backspace}');
+      expect(screen.getByLabelText('the label')).toHaveValue('');
+      expect(screen.getByLabelText('the label')).toBeInvalid();
+      expect(screen.getByLabelText('the label')).toHaveDescription('Constraints not satisfied');
```

Comment:
This assertion might now be wrong with the new validation spec. Consult your friendly designer.